org.eclipse.jdt.core/batch/org/eclipse/jdt/internal/compiler/batch/ClasspathJar.java

```diff
@@ -285,15 +312,12 @@ protected void addToPackageCache(String fileName, boolean endsWithSep) {
 public synchronized char[][] getModulesDeclaringPackage(String qualifiedPackageName, String moduleName) {
 	if (this.packageCache != null)
 		return singletonModuleNameIf(this.packageCache.contains(qualifiedPackageName));
-
 	// AspectJ Extension
 	try {
-        ensureOpen();
-	} catch (IOException ioe) {
-		// Doesn't normally occur - probably means since starting the compile
-		// you have removed one of the jars.
-		ioe.printStackTrace();
-		return singletonModuleNameIf(false);
+		ensureOpen();
+	}
+	catch (IOException e) {
+		throw new RuntimeException(e);
```

Comment:
this one too is a change in behaviour isn't it?

Reply:
Yes. Same rationale as above.